frontend/src/pages/projects/screens/detail/storage/StorageTable.tsx

```diff
@@ -15,20 +20,65 @@ type StorageTableProps = {
 const StorageTable: React.FC<StorageTableProps> = ({ pvcs, refresh, onAddPVC }) => {
   const [deleteStorage, setDeleteStorage] = React.useState<PersistentVolumeClaimKind | undefined>();
   const [editPVC, setEditPVC] = React.useState<PersistentVolumeClaimKind | undefined>();
+  const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
+  const [storageClasses, storageClassesLoaded] = useStorageClasses();
+  const [alertDismissed, setAlertDismissed] = React.useState<boolean>(false);
+  const storageTableData: StorageTableData[] = pvcs.map((pvc) => ({
+    pvc,
+    storageClass: storageClasses.find((sc) => sc.metadata.name === pvc.spec.storageClassName),
+  }));
+  const isDeprecatedAlert = React.useMemo(
+    () =>
+      storageClassesLoaded &&
+      storageTableData.some(
+        (data) => !data.storageClass || !getStorageClassConfig(data.storageClass)?.isEnabled,
+      ),
+    [storageClassesLoaded, storageTableData],
+  );
+  const shouldShowAlert = isDeprecatedAlert && !alertDismissed && isStorageClassesAvailable;
+
+  const getStorageColumns = () => {
+    let storageColumns = columns;
+
+    if (!isStorageClassesAvailable) {
+      storageColumns = columns.filter((column) => column.field !== 'storage');
+    }
+    return storageColumns;
+  };
 
   return (
     <>
+      {shouldShowAlert && (
+        <Alert
+          data-testid="storage-class-deprecated-alert"
+          variant="warning"
+          isInline
+          title="Deprecated storage class"
+          actionClose={
+            <AlertActionCloseButton
+              data-testid="storage-class-deprecated-alert-close-button"
+              onClose={() => setAlertDismissed(true)}
+            />
+          }
+        >
+          A storage class has been deprecated by your administrator, but the cluster storage using
+          it is still active. If you want to migrate your data to cluster storage instance using a
+          different storage class, contact your administrator.
+        </Alert>
+      )}
       <Table
-        data={pvcs}
-        columns={columns}
+        data={storageTableData}
+        columns={getStorageColumns()}
         disableRowRenderSupport
         data-testid="storage-table"
         variant="compact"
-        rowRenderer={(pvc, i) => (
+        rowRenderer={(data, i) => (
           <StorageTableRow
-            key={pvc.metadata.uid}
+            key={data.pvc.metadata.uid}
             rowIndex={i}
-            obj={pvc}
+            storageClass={data.storageClass}
+            storageClassesLoaded={storageClassesLoaded}
+            pvc={data.pvc}
```

Comment:
you dont need to pass in storageClass or storageClassLoaded

data already has the storage class so just pass in `obj={data}`

Reply:
The `storageClassesLoaded` must be passed because when the `storageClass` is in a loading state, it is undefined, which can be misinterpreted as a `deleted storageClass` for fraction of second. The `storageClassesLoaded` value comes from const [storageClasses, storageClassesLoaded] = useStorageClasses().

<img width="1134" alt="Screenshot 2024-09-20 at 11 14 05 AM" src="https://github.com/user-attachments/assets/22499624-11a3-4ee3-9c18-81372c071e42">

I have made the first changes